src/Mod/Assembly/App/AssemblyLink.cpp

```diff
@@ -207,14 +215,18 @@ void AssemblyLink::synchronizeComponents()
 
             auto* subAsmLink = freecad_cast<AssemblyLink*>(obj2);
             auto* link2 = dynamic_cast<App::Link*>(obj2);
+
             if (subAsmLink) {
                 linkedObj = subAsmLink->getLinkedObject2(false);  // not recursive
             }
             else if (link2) {
                 linkedObj = link2->getLinkedObject(false);  // not recursive
             }
+            // else if (group) {
+            //     linkedObj = group->getLinkedObject(false);  // not recursive
+            // }
             else {
-                // We consider only Links and AssemblyLinks in the AssemblyLink.
+                // We consider only Links, AssemblyLinks, and in the AssemblyLink.
```

Comment:
This edits the comment but new version does not make sense. There is something missing after 'and' also you don't need a ',' before a 'and'